pkg/common/prometheus/metrics.go

```diff
@@ -42,6 +42,24 @@ const (
 	// PrometheusExpandVolumeOpType represents the ExpandVolume operation.
 	PrometheusExpandVolumeOpType = "expand-volume"
 
+	// CSI operation fault types
+
+	// PrometheusFaultInvalidArgumentType represents the InvalidArgument fault
+	// that CSI operation may failed with
+	PrometheusFaultInvalidArgumentType = "InvalidArgument"
+	// PrometheusFaultNotFoundType represents the NotFound fault
+	// that CSI operation may failed with
+	PrometheusFaultNotFoundType = "NotFound"
+	// PrometheusFaultResourceInUseType represents the ResourceInUse fault
+	// that CSI operation may failed with
+	PrometheusFaultResourceInUseType = "ResourceInUse"
+	// PrometheusFaultManagedObjectNotFoundType represents the ManagedObjectNotFound fault
+	// that CSI operation may failed with
+	PrometheusFaultManagedObjectNotFoundType = "ManagedObjectNotFound"
+	// PrometheusFaultOtherFaultType represents the fault
+	// that CSI operation may failed with other than the fault listed above
+	PrometheusFaultOtherFaultType = "OtherFault"
+
```

Comment:
I am worried about having to enumerate all fault types like this. Most likely CNS will add new fault types over time, and it is a maintenance issue to have to update this list every time that happens

Reply:
+1 to this comment, that was my first thought after looking at the approach followed in this PR.